## Challenge 1 - Independent Sample T-tests

In this challenge, we will be using the Pokemon dataset. Before applying statistical methods to this data, let's first examine the data.

To load the data, run the code below.


In [1]:
import pandas as pd
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [2]:
# Run this code:

pokemon = pd.read_csv('./pokemon.csv')

Let's start off by looking at the head function in the cell below.


In [3]:
pokemon.head()

#                   Name Type 1  Type 2  Total  HP  Attack  Defense  \
0  1              Bulbasaur  Grass  Poison    318  45      49       49   
1  2                Ivysaur  Grass  Poison    405  60      62       63   
2  3               Venusaur  Grass  Poison    525  80      82       83   
3  3  VenusaurMega Venusaur  Grass  Poison    625  80     100      123   
4  4             Charmander   Fire     NaN    309  39      52       43   

   Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0       65       65     45           1      False  
1       80       80     60           1      False  
2      100      100     80           1      False  
3      122      120     80           1      False  
4       60       50     65           1      False

The first thing we would like to do is compare the legendary Pokemon to the regular Pokemon. To do this, we should examine the data further. What is the count of legendary vs. non legendary Pokemons?

In [4]:
legend = pokemon.groupby('Legendary').count()['Name'].reset_index()
legend

Legendary  Name
0      False   735
1       True    65

Compute the mean and standard deviation of the total points for both legendary and non-legendary Pokemon.



In [6]:
tp_leg = pokemon[pokemon['Legendary'] == True]['Total']

print(tp_leg.mean())
print(tp_leg.std())

637.3846153846154
60.93738905315346


In [7]:
tp_non_leg = pokemon[pokemon['Legendary'] == False]['Total']

print(tp_non_leg.mean())
print(tp_non_leg.std())

417.21360544217686
106.76041745713022


The computation of the mean might give us a clue regarding how the statistical test may turn out; However, it certainly does not prove whether there is a significant difference between the two groups.

In the cell below, use the ttest_ind function in scipy.stats to compare the total points for legendary and non-legendary Pokemon. Since we do not have any information about the population, assume the variances are not equal.

In [8]:
stats.ttest_ind(a=tp_leg, b=tp_non_leg, equal_var=False)

Ttest_indResult(statistic=25.8335743895517, pvalue=9.357954335957446e-47)

What do you conclude from this test? Write your conclusions below.



**Since the p value (9.357954335957446e-47) is lower than alpha (0.05), we reject the null hypothesis (H0) which states that there is no significant difference between the pokemons that are Legendary and the ones who are not. An alternate hypothesis, the one we accept, states that there is a significant difference between the two.**

How about we try to compare the different types of pokemon? In the cell below, list the types of Pokemon from column Type 1 and the count of each type.



In [9]:
pokemon['Type 1'].value_counts()

Water       112
Normal       98
Grass        70
Bug          69
Psychic      57
Fire         52
Electric     44
Rock         44
Dragon       32
Ground       32
Ghost        32
Dark         31
Poison       28
Steel        27
Fighting     27
Ice          24
Fairy        17
Flying        4
Name: Type 1, dtype: int64

Since water is the largest group of Pokemon, compare the mean and standard deviation of water Pokemon to all other Pokemon.



In [10]:
water_poke = pokemon[pokemon['Type 1'] == 'Water']['Total']
print(water_poke.mean())
print(water_poke.std())

430.45535714285717
113.18826606431458


In [11]:
non_water_poke = pokemon[pokemon['Type 1'] != 'Water']['Total']
print(non_water_poke.mean())
print(non_water_poke.std())

435.85901162790697
121.09168230208066


Perform a hypothesis test comparing the mean of total points for water Pokemon to all non-water Pokemon. Assume the variances are equal.



In [12]:
stats.ttest_ind(a=water_poke, b=non_water_poke, equal_var=True)

Ttest_indResult(statistic=-0.4418547448849676, pvalue=0.6587140317488793)

Write your conclusion below.



**Since the p value (0.6587140317488793) is greater than alpha (0.05), we accept the null hypothesis (H0) which states that there is no significant difference between the water pokemons ans the non water pokemons.**

## Challenge 2 - Matched Pairs Test

In this challenge we will compare dependent samples of data describing our Pokemon. Our goal is to see whether there is a significant difference between each Pokemon's defense and attack scores. Our hypothesis is that the defense and attack scores are equal. In the cell below, import the ttest_rel function from scipy.stats and compare the two columns to see if there is a statistically significant difference between them.



In [18]:
stats.ttest_rel(pokemon['Attack'], pokemon['Defense'])

Ttest_relResult(statistic=4.325566393330478, pvalue=1.7140303479358558e-05)

Describe the results of the test in the cell below.



**Since the p value (91.7140303479358558e-05) is lower than alpha (0.05), we accept the alternate hypothesis, which states that there is a significant difference between each Pokemon's defense and attack scores, and they're not equal.**

We are also curious about whether there is a significant difference between the mean of special defense and the mean of special attack. Perform the hypothesis test in the cell below.



In [14]:
stats.ttest_rel(pokemon['Sp. Def'], pokemon['Sp. Atk'])

Ttest_relResult(statistic=-0.853986188453353, pvalue=0.3933685997548122)

Describe the results of the test in the cell below.



**Since the p value (0.3933685997548122) is greater than alpha (0.05), we accept the null hypothesis (H0) which states that there is no significant difference between the mean of special defense and the mean of special attack.**

As you may recall, a two sample matched pairs test can also be expressed as a one sample test of the difference between the two dependent columns.

Import the ttest_1samp function and perform a one sample t-test of the difference between defense and attack. Test the hypothesis that the difference between the means is zero. Confirm that the results of the test are the same.

In [15]:
stats.ttest_1samp(pokemon['Defense'] - pokemon['Attack'], popmean=0)

Ttest_1sampResult(statistic=-4.325566393330478, pvalue=1.7140303479358558e-05)

**Since the p value (1.7140303479358558e-05) is lower than alpha (0.05), we reject the null hypothesis and instead accept the alternate one, which states that the difference between the defense and attack variables is not zero.**

## Bonus Challenge - The Chi-Square Test

The Chi-Square test is used to determine whether there is a statistically significant difference in frequencies. In other words, we are testing whether there is a relationship between categorical variables or rather when the variables are independent. This test is an alternative to Fisher's exact test and is used in scenarios where sample sizes are larger. However, with a large enough sample size, both tests produce similar results. Read more about the Chi Squared test here.

In the cell below, create a contingency table using pd.crosstab comparing whether a Pokemon is legendary or not and whether the Type 1 of a Pokemon is water or not.



In [21]:
leg_wat = pd.crosstab(index=pokemon['Legendary'], columns=pokemon['Type 1'].eq('Water'))
leg_wat

Type 1     False  True
Legendary             
False        627   108
True          61     4

Perform a chi-squared test using the chi2_contingency function in scipy.stats. 



In [22]:
stat, p_value, dof, expected = chi2_contingency(leg_wat)

print(stat)
print(p_value)
print(dof)
print(expected)

2.9429200762850503
0.08625467249550949
1
[[632.1 102.9]
 [ 55.9   9.1]]


Based on a 95% confidence, should we reject the null hypothesis?



**Since the p value (0.08625467249550949) is greater than alpha (0.05), we accept the null hypothesis (H0) which states that there is no significant relationship between the variables Legendary and Water Type. An alternate hypothesis would state that there is a significant relation between the two.**